src/rgw/rgw_bucket.cc

```diff
@@ -295,29 +295,28 @@ int rgw_bucket_instance_remove_entry(RGWRados *store, string& entry, RGWObjVersi
   return store->meta_mgr->remove_entry(bucket_instance_meta_handler, entry, objv_tracker);
 }
 
+// parse instance in format: [tenant:]instance[:shard_id], returning
```

Comment:
@cbodley this is ambiguous. can be either tenant:instance, or instance:shard_id. Will only work if we're guaranteed that instance does not parse as integer (and shard_id does). Do we want to tie our hands like that?